org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/tasks/TaskMarkerContributor.xtend

```diff
@@ -43,10 +44,39 @@ class TaskMarkerContributor implements IMarkerContributor {
 			if (monitor.isCanceled()) {
 				throw new OperationCanceledException();
 			}
-			deleteMarkers(file, monitor);
-			createTaskMarkers(file, tasks, monitor);
+			val markers = file.findMarkers(TaskMarkerTypeProvider.XTEXT_TASK_TYPE, true, IResource.DEPTH_ZERO);
+			
+			// Avoid recreation of existing markers. This can result in a build loop.
+			val hasDelta = tasks.exists[t|!markers.exists[m|match(t,m)]] || markers.exists[m|!tasks.exists[t|match(t,m)]]
+			if (hasDelta) {
+				deleteMarkers(file, monitor);
+				createTaskMarkers(file, tasks, monitor);
+			}
 		} catch (CoreException e) {
-			log.error(e.getMessage(), e);
+			LOG.error(e.getMessage(), e);
+		}
+	}
+	
+	private def boolean match (Task task, IMarker marker) {
+		marker.getAttribute(IMarker.LOCATION) == "line " + task.lineNumber
```

Comment:
Just in case you have to change something else too; you could move down this to the very end right after `CHAR_END` and pull `USER_EDITABLE` up. The first check concatenates a string with an integer which might be unnecessary... I know this is extremely minor performance tweak.

Reply:
Ahh, ignore my comment and sorry for the noise.